src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -157,11 +161,70 @@ protected function addRoute(RouteCollection $collection, $annot, $globals, \Refl
             $condition = $globals['condition'];
         }
 
-        $route = $this->createRoute($globals['path'].$annot->getPath(), $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+        $path = $annot->getPath();
+        $locales = $annot->getLocales();
+        $hasLocalizedPrefix = !empty($globals['locales']);
+        $hasPrefix = $hasLocalizedPrefix || !empty($globals['path']);
```

Comment:
I would really avoid using `empty` whenever possible. `$globals['path'] = '0';` would be considered empty here, which is not what we want.

Reply:
Converted this to `null ===` checks to prevent this.